[hail] bad error message in import_matrix_table

```
# cat foo
7	75216143	75216143	C/T	+
# python <<EOF
import hail as hl
hl.import_matrix_table(
    'foo',
    row_fields={f'f{i}': hl.tstr for i in range(5)},
    no_header=True).count()
EOF
```
produces:
```
Hail version: 0.2.11-cf54f08305d1
Error summary: HailException: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
```
I expect it to say something like "yo dawg, you forgot to have entries, maybe you actually want import_table".

full output:
```
Initializing Spark and Hail with default parameters...
using hail jar at /Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/hail/hail-all-spark.jar
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
19/03/27 18:27:22 WARN Utils: Your hostname, wmb16-359 resolves to a loopback address: 127.0.0.1; using 10.1.1.163 instead (on interface en0)
19/03/27 18:27:22 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
Running on Apache Spark version 2.2.3
SparkUI available at http://10.1.1.163:4041
Welcome to
     __  __     <>__
    / /_/ /__  __/ /
   / __  / _ `/ / /
  /_/ /_/\_,_/_/_/   version 0.2.11-cf54f08305d1
LOGGING: writing to /Users/dking/hail-20190327-1827-0.2.11-cf54f08305d1.log
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "/Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/hail/matrixtable.py", line 2371, in count
    return (self.count_rows(), self.count_cols())
  File "/Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/hail/matrixtable.py", line 2331, in count_rows
    TableCount(MatrixRowsTable(self._mir)))
  File "/Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/hail/backend/backend.py", line 94, in execute
    self._to_java_ir(ir)))
  File "/Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/Users/dking/anaconda2/envs/hail/lib/python3.6/site-packages/hail/utils/java.py", line 227, in deco
    'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
hail.utils.java.FatalError: HailException: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+

Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 1, localhost, executor driver): is.hail.utils.HailException: foo: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
  offending line: 7	75216143	75216143	C/T	+
	at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:20)
	at is.hail.utils.package$.fatal(package.scala:28)
	at is.hail.utils.Context.wrapException(Context.scala:19)
	at is.hail.utils.WithContext.wrap(Context.scala:43)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:377)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:375)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$12.next(Iterator.scala:444)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:575)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:573)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.fold(TraversableOnce.scala:212)
	at scala.collection.AbstractIterator.fold(Iterator.scala:1336)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:344)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: is.hail.utils.HailException: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
	at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:9)
	at is.hail.utils.package$.fatal(package.scala:28)
	at is.hail.io.LoadMatrixParser.parseLine(LoadMatrix.scala:33)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6$$anonfun$apply$7.apply(LoadMatrix.scala:383)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6$$anonfun$apply$7.apply(LoadMatrix.scala:377)
	at is.hail.utils.WithContext.wrap(Context.scala:41)
	... 30 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1575)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1563)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1562)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1562)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:803)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:803)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:803)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1790)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1745)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1734)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:619)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2029)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1.apply(RDD.scala:1097)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
	at org.apache.spark.rdd.RDD.fold(RDD.scala:1091)
	at is.hail.rvd.RVD.count(RVD.scala:580)
	at is.hail.expr.ir.Interpret$$anonfun$apply$1.apply$mcJ$sp(Interpret.scala:756)
	at is.hail.expr.ir.Interpret$$anonfun$apply$1.apply(Interpret.scala:756)
	at is.hail.expr.ir.Interpret$$anonfun$apply$1.apply(Interpret.scala:756)
	at scala.Option.getOrElse(Option.scala:121)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:756)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:86)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:58)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$7.apply(InterpretNonCompilable.scala:19)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$7.apply(InterpretNonCompilable.scala:19)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at is.hail.expr.ir.InterpretNonCompilable$.apply(InterpretNonCompilable.scala:19)
	at is.hail.expr.ir.CompileAndEvaluate$.apply(CompileAndEvaluate.scala:39)
	at is.hail.expr.ir.CompileAndEvaluate$.evaluateToJSON(CompileAndEvaluate.scala:14)
	at is.hail.expr.ir.CompileAndEvaluate.evaluateToJSON(CompileAndEvaluate.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:745)

is.hail.utils.HailException: foo: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
  offending line: 7	75216143	75216143	C/T	+
	at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:20)
	at is.hail.utils.package$.fatal(package.scala:28)
	at is.hail.utils.Context.wrapException(Context.scala:19)
	at is.hail.utils.WithContext.wrap(Context.scala:43)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:377)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:375)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$12.next(Iterator.scala:444)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:575)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:573)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.fold(TraversableOnce.scala:212)
	at scala.collection.AbstractIterator.fold(Iterator.scala:1336)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:344)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

is.hail.utils.HailException: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
	at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:9)
	at is.hail.utils.package$.fatal(package.scala:28)
	at is.hail.io.LoadMatrixParser.parseLine(LoadMatrix.scala:33)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6$$anonfun$apply$7.apply(LoadMatrix.scala:383)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6$$anonfun$apply$7.apply(LoadMatrix.scala:377)
	at is.hail.utils.WithContext.wrap(Context.scala:41)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:377)
	at is.hail.io.LoadMatrix$$anonfun$16$$anonfun$apply$6.apply(LoadMatrix.scala:375)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$12.next(Iterator.scala:444)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:575)
	at is.hail.rvd.RVD$$anonfun$count$2.apply(RVD.scala:573)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitions$1$$anonfun$apply$28.apply(ContextRDD.scala:405)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at is.hail.sparkextras.ContextRDD$$anonfun$run$1$$anonfun$apply$8.apply(ContextRDD.scala:192)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.fold(TraversableOnce.scala:212)
	at scala.collection.AbstractIterator.fold(Iterator.scala:1336)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.rdd.RDD$$anonfun$fold$1$$anonfun$20.apply(RDD.scala:1095)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$33.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:344)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)





Hail version: 0.2.11-cf54f08305d1
Error summary: HailException: Error parsing row fields in row 0:
    expected 5 fields but only 5 found.
    File: foo
    Line:
        7	75216143	75216143	C/T	+
```

--- a/hail/python/test/hail/methods/test_impex.py
+++ b/hail/python/test/hail/methods/test_impex.py
@@ -1229,7 +1229,7 @@ class ImportMatrixTableTests(unittest.TestCase):
         mt = hl.import_matrix_table(file, row_fields=fields, row_key=['Chromosome', 'Position'])
         t = hl.import_table(file, types=fields, key=['Chromosome', 'Position'])
 
-        self.assertEquals(mt.count_cols(), 0)
+        self.assertEqual(mt.count_cols(), 0)
         self.assertTrue(t._same(mt.rows()))
 
 

--- a/hail/python/test/hail/methods/test_impex.py
+++ b/hail/python/test/hail/methods/test_impex.py
@@ -1222,6 +1222,10 @@ class ImportMatrixTableTests(unittest.TestCase):
                           no_header=True,
                           row_key=['foo'])
 
+    @skip_unless_spark_backend()
+    def test_import_matrix_table_no_cols(self):
+        mt = hl.import_matrix_table(resource('sample2_va_nomulti.tsv'))
+
 
 class ImportTableTests(unittest.TestCase):
     def test_import_table_force_bgz(self):

--- a/hail/python/test/hail/methods/test_impex.py
+++ b/hail/python/test/hail/methods/test_impex.py
@@ -1224,7 +1224,13 @@ class ImportMatrixTableTests(unittest.TestCase):
 
     @skip_unless_spark_backend()
     def test_import_matrix_table_no_cols(self):
-        mt = hl.import_matrix_table(resource('sample2_va_nomulti.tsv'))
+        fields = { 'Chromosome':hl.str, 'Position': hl.tint32, 'Ref': hl.tstr, 'Alt': hl.tstr, 'Rand1': hl.tfloat64, 'Rand2':hl.tfloat64 }
+        file = resource('sample2_va_nomulti.tsv')
+        mt = hl.import_matrix_table(file, row_fields=fields, row_key=['Chromosome', 'Position'])
+        t = hl.import_table(file, types=fields, key=['Chromosome', 'Position'])
+
+        self.assertEquals(mt.count_cols(), 0)
+        self.assertTrue(t._same(mt.rows()))
 
 
 class ImportTableTests(unittest.TestCase):

--- a/hail/python/test/hail/methods/test_impex.py
+++ b/hail/python/test/hail/methods/test_impex.py
@@ -1224,7 +1224,7 @@ class ImportMatrixTableTests(unittest.TestCase):
 
     @skip_unless_spark_backend()
     def test_import_matrix_table_no_cols(self):
-        fields = { 'Chromosome':hl.str, 'Position': hl.tint32, 'Ref': hl.tstr, 'Alt': hl.tstr, 'Rand1': hl.tfloat64, 'Rand2':hl.tfloat64 }
+        fields = {'Chromosome':hl.tstr, 'Position': hl.tint32, 'Ref': hl.tstr, 'Alt': hl.tstr, 'Rand1': hl.tfloat64, 'Rand2': hl.tfloat64}
         file = resource('sample2_va_nomulti.tsv')
         mt = hl.import_matrix_table(file, row_fields=fields, row_key=['Chromosome', 'Position'])
         t = hl.import_table(file, types=fields, key=['Chromosome', 'Position'])

--- a/hail/src/main/scala/is/hail/io/LoadMatrix.scala
+++ b/hail/src/main/scala/is/hail/io/LoadMatrix.scala
@@ -35,9 +35,9 @@ class LoadMatrixParser(rvb: RegionValueBuilder, fieldTypes: Array[Type], entryTy
              |    Line:
              |        ${ line.truncate }""".stripMargin
         )
+      }
       off = addType(fieldTypes(ii))(line, rowNum, ii, off)
       ii += 1
-      }
     }
 
     ii = 0
@@ -209,7 +209,7 @@ object LoadMatrix {
       lines match {
         case Array(header, first) =>
           val nCols = first.split(charRegex(sep), -1).length - nRowFields
-          if (nCols <= 0)
+          if (nCols < 0)
             fatal(s"More row fields ($nRowFields) than columns (${ nRowFields + nCols }) in file: $file")
           (header.split(charRegex(sep), -1), nCols)
         case _ =>

--- a/hail/src/main/scala/is/hail/io/LoadMatrix.scala
+++ b/hail/src/main/scala/is/hail/io/LoadMatrix.scala
@@ -27,8 +27,6 @@ class LoadMatrixParser(rvb: RegionValueBuilder, fieldTypes: Array[Type], entryTy
     var ii = 0
     var off = 0
     while (ii < fieldTypes.length) {
-      off = addType(fieldTypes(ii))(line, rowNum, ii, off)
-      ii += 1
       if (off > line.length) {
         fatal(
           s"""Error parsing row fields in row $rowNum:
@@ -37,6 +35,8 @@ class LoadMatrixParser(rvb: RegionValueBuilder, fieldTypes: Array[Type], entryTy
              |    Line:
              |        ${ line.truncate }""".stripMargin
         )
+      off = addType(fieldTypes(ii))(line, rowNum, ii, off)
+      ii += 1
       }
     }
 

